enterprise/lms_api.py

```diff
@@ -114,7 +119,79 @@ def get_course_details(self, course_id):
         Returns:
             dict: Contains keys identifying those course details available from the courses API (e.g., name).
         """
-        return self.client.courses(course_id).get()
+        try:
+            return self.client.courses(course_id).get()
+        except HttpNotFoundError:
+            LOGGER.error('details not found for course=%s', course_id)
+            return None
+
+
+class ThirdPartyAuthApiClient(LmsApiClient):
+    """
+    Object builds an API client to make calls to the Third Party Auth API.
+    """
+
+    API_BASE_URL = settings.LMS_ROOT_URL + '/api/third_party_auth/v0/'
+
+    def get_remote_id(self, identity_provider, username):
+        """
+        Retrieve the remote identifier for the given username.
+
+        Args:
+        * ``identity_provider`` (str): identifier slug for the third-party authentication service used during SSO.
+        * ``username`` (str): The username ID identifying the user for which to retrieve the remote name.
+
+        Returns:
+            string or None: the remote name of the given user.  None if not found.
+        """
+        try:
+            returned = self.client.providers(identity_provider).users(username).get()
+            results = returned.get('results', [])
+        except HttpNotFoundError:
+            LOGGER.error('remote_id not found for third party provider=%s, username=%s', identity_provider, username)
+            results = []
+
+        for row in results:
+            if row.get('username') == username:
+                return row.get('remote_id')
+        return None
+
+
+class GradesApiClient(LmsApiClient):
+    """
+    Object builds an API client to make calls to the LMS Grades API.
+    """
+
+    API_BASE_URL = settings.LMS_ROOT_URL + '/api/grades/v0/'
+    APPEND_SLASH = True
+
+    def get_course_grade(self, course_id, username):
+        """
+        Retrieve the grade for the given username for the given course_id.
+
+        Args:
+        * ``course_id`` (str): The string value of the course's unique identifier
+        * ``username`` (str): The username ID identifying the user for which to retrieve the grade.
+
+        Returns: None if no grade found, or a dict containing:
+
+        * ``username``: A string representation of a user's username passed in the request.
+        * ``course_key``: A string representation of a Course ID.
+        * ``passed``: Boolean representing whether the course has been passed according the course's grading policy.
+        * ``percent``: A float representing the overall grade for the course
+        * ``letter_grade``: A letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None
+
+        """
+        try:
+            results = self.client.course_grade(course_id).users().get(username=username)
+        except HttpNotFoundError:
```

Comment:
@pomegranited suggestion: this `try .. except: log.error(...) results = []` repeats a couple of times, how about extracting it into method? Might be not worth it, so please use common sense :)